programs/helium-entity-manager/src/instructions/genesis_issue_hotspot_v0.rs

```diff
@@ -0,0 +1,146 @@
+  #[account(mut)]
+  pub collection_metadata: UncheckedAccount<'info>,
+  /// CHECK: Handled by cpi
+  pub collection_master_edition: UncheckedAccount<'info>,
+  #[account(
+    has_one = collection,
+    has_one = merkle_tree
+  )]
+  pub hotspot_config: Box<Account<'info, HotspotConfigV0>>,
+  #[account(
+    init,
+    payer = lazy_signer,
+    space = 8 + 60 + std::mem::size_of::<HotspotStorageV0>(),
+    seeds = [
+      "storage".as_bytes(),
+      &args.ecc_compact
+    ],
+    bump
+  )]
+  pub storage: Box<Account<'info, HotspotStorageV0>>,
+  /// CHECK: Handled by cpi
+  #[account(mut)]
+  pub tree_authority: AccountInfo<'info>,
+  /// CHECK: Used in cpi
+  pub recipient: AccountInfo<'info>,
+  /// CHECK: Used in cpi
+  #[account(mut)]
+  pub merkle_tree: AccountInfo<'info>,
+  #[account(
+    seeds = ["collection_cpi".as_bytes()],
+    seeds::program = bubblegum_program.key(),
+    bump,
+  )]
+  /// CHECK: Used in cpi
+  pub bubblegum_signer: UncheckedAccount<'info>,
+
+  /// CHECK: Verified by constraint  
+  #[account(address = mpl_token_metadata::ID)]
+  pub token_metadata_program: AccountInfo<'info>,
+  pub log_wrapper: Program<'info, Wrapper>,
+  pub bubblegum_program: Program<'info, Bubblegum>,
+  pub compression_program: Program<'info, SplAccountCompression>,
+  pub system_program: Program<'info, System>,
+  pub rent: Sysvar<'info, Rent>,
+}
+
+impl<'info> GenesisIssueHotspotV0<'info> {
+  fn mint_to_collection_ctx(&self) -> CpiContext<'_, '_, '_, 'info, MintToCollectionV1<'info>> {
+    let cpi_accounts = MintToCollectionV1 {
+      tree_authority: self.tree_authority.to_account_info(),
+      leaf_delegate: self.recipient.to_account_info(),
+      leaf_owner: self.recipient.to_account_info(),
+      merkle_tree: self.merkle_tree.to_account_info(),
+      payer: self.lazy_signer.to_account_info(),
+      tree_delegate: self.hotspot_config.to_account_info(),
+      log_wrapper: self.log_wrapper.to_account_info(),
+      compression_program: self.compression_program.to_account_info(),
+      system_program: self.system_program.to_account_info(),
+      collection_authority: self.hotspot_config.to_account_info(),
+      collection_authority_record_pda: self.bubblegum_program.to_account_info(),
+      collection_mint: self.collection.to_account_info(),
+      collection_metadata: self.collection_metadata.to_account_info(),
+      edition_account: self.collection_master_edition.to_account_info(),
+      bubblegum_signer: self.bubblegum_signer.to_account_info(),
+      token_metadata_program: self.token_metadata_program.to_account_info(),
+    };
+    CpiContext::new(self.bubblegum_program.to_account_info(), cpi_accounts)
+  }
+}
+
+pub fn handler(ctx: Context<GenesisIssueHotspotV0>, args: GenesisIssueHotspotArgsV0) -> Result<()> {
+  let decoded = bs58::encode(args.ecc_compact.clone()).into_string();
+  let animal_name: AnimalName = decoded
+    .parse()
+    .map_err(|_| error!(ErrorCode::InvalidEccCompact))?;
```

Comment:
Do we do this logic else where besides `genesis_issue` and `issue`? Might be nice to move it to shared-utils

Reply:
Should only be these two places. And genesis issue hotspot hopefully disappears eventually